src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs

```diff
@@ -145,12 +147,18 @@ public async Task StartAsync(string target = null)
                 _serviceConnectionStartTcs.TrySetResult(true);
                 try
                 {
+                    var syncTimer = new TimerAwaitable(TimeSpan.Zero, DefaultSyncAzureIdentityInterval);
```

Comment:
you don't dispose the timer if it is not aadKey.

Can we prevent creating the timer if it is not aadKey?